cfme/scripting/quickstart.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import print_function
+
+import os
+import sys
+import argparse
+import subprocess
+import hashlib
+
+try:
+    import distro
+except ImportError:
+    sys.path.insert(0, os.path.join(
+        os.path.dirname(os.path.dirname(
+            os.path.dirname(__file__))),
+        'requirements', 'quickstart',
+        'distro-1.0.4-py2.py3-none-any.whl'))
+    import distro
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--mk_virtualenv", default="../cfme_venv")
+parser.add_argument("--system-site-packages", action="store_true")
+parser.add_argument("--config_path", default="../cfme-qe-yamls/complete/")
+
+DISTRO_DATA = distro.linux_distribution()[:2]
+IS_SCRIPT = sys.argv[0] == __file__
+CWD = os.getcwd()  # we expect to be in the workdir
+IS_ROOT = os.getuid() == 0
+REDHAT_BASED = os.path.isfile('/etc/redhat-release')
+CREATED = object()
+REQUIREMENT_FILE = 'requirements/frozen.txt'
+HAS_DNF = os.path.exists('/usr/bin/dnf')
+IN_VIRTUALENV = getattr(sys, 'real_prefix', None) is not None
+
+PRISTINE_ENV = dict(os.environ)
+
+REDHAT_PACKAGES_OLD = (
+    " python-virtualenv gcc postgresql-devel libxml2-devel"
+    " libxslt-devel zeromq3-devel libcurl-devel"
+    " redhat-rpm-config gcc-c++ openssl-devel"
+    " libffi-devel python-devel")
+
+
+REDHAT_PACKAGES_F25 = (
+    " python2-virtualenv gcc postgresql-devel libxml2-devel"
+    " libxslt-devel zeromq3-devel libcurl-devel"
+    " redhat-rpm-config gcc-c++ openssl-devel"
+    " libffi-devel python2-devel")
+
+
+if REDHAT_BASED:
+    os.environ['PYCURL_SSL_LIBRARY'] = 'nss'
+    if DISTRO_DATA == ("Fedora", "25"):
+        REDHAT_PACKAGES = REDHAT_PACKAGES_F25
+    else:
+        REDHAT_PACKAGES = REDHAT_PACKAGES_OLD
+
+    if HAS_DNF:
+        INSTALL_COMMAND = 'dnf install -y'
+    else:
+        INSTALL_COMMAND = 'yum install -y'
+    if not IS_ROOT:
+        INSTALL_COMMAND = 'sudo ' + INSTALL_COMMAND
+
+
+def command_text(command, shell):
+    if shell:
+        return command
+    else:
+        command = [repr(x) if ' ' in x else x for x in command]
```

Comment:
@mfalesni utils is not importable because we put all the things into a toplevel  __init__